website/client/src/components/auth/socialAuth/apple.vue

```diff
@@ -0,0 +1,109 @@
+<template>
+  <div
+    :class="{
+      'btn': !isHomePage,
+      'btn-secondary': !isHomePage,
+      'social-button': true,
+      'social-button-home': isHomePage,
+    }"
+    @click="socialAuth('apple')"
+  >
+    <div
+      class="svg-icon social-icon apple-icon"
+      v-html="icons.appleIcon"
+    ></div>
+    <span>{{ (registering || isHomePage || isGroupPlansPage)
+      ? $t('signUpWithSocial', {social: 'Apple'})
+      : $t('loginWithSocial', {social: 'Apple'}) }}</span>
+  </div>
+</template>
+
+<style lang="scss">
+  .social-button {
+    width: 100%;
+    height: 100%;
+    white-space: inherit;
+    text-align: center;
+
+    .text {
+      display: inline-block;
+    }
+  }
+
+  .social-button-home {
+    border-radius: 2px;
+    border: solid 2px #bda8ff;
+    width: 100%;
+    min-height: 40px;
+    padding: .5em;
+    background: transparent;
+    margin-bottom: .5em;
+    color: #bda8ff;
+    transition: .5s;
+
+    span {
+      transition: none;
+    }
+  }
+
+  .social-button-home:hover {
+    cursor: pointer;
+    border-color: #fff;
+    color: #fff;
+  }
+
+  .apple-icon {
+    margin-top: -1px;
+  }
+
+  .social-icon {
+    margin-left: 1em;
+    margin-right: 1em;
+    width: 18px;
+    height: 18px;
+    display: inline-block;
+    vertical-align: top;
+    margin-top: .1em;
+  }
+</style>
+
+<script>
+import { buildAppleAuthUrl } from '@/libs/auth';
+import appleIcon from '@/assets/svg/apple_black.svg';
+
+export default {
+  data () {
+    const data = {};
+    data.icons = Object.freeze({
+      appleIcon,
+    });
+    return data;
+  },
+  computed: {
+    registering () {
+      if (this.$route.path.startsWith('/register')) {
```

Comment:
Continuing on this path, what the code here needs to know is not "where it is shown", but it needs to now what it needs to do. "Should it show the signup instead of the login?" "Should it show the fancy graphics on the button?".

So then ideally the inclusion would be something like this: `<apple-auth show-sign-up looks-fancy/>`. (The properties don't need a value since boolean properties default to true when they're present)

There's probably a better name for `looks-fancy`, but you get the gist. This way you keep the API of the component about what this component can do and you fully decouple it from where it is used.